elastic4s-effect-cats-2/src/test/scala/com.sksamuel.elastic4s.cats.effect/CatsEffectTest.scala

```diff
@@ -0,0 +1,36 @@
+package com.sksamuel.elastic4s.cats.effect
+
+import cats.data.OptionT
+import cats.effect.IO
+import com.sksamuel.elastic4s.{ElasticClient, ElasticProperties}
+import com.sksamuel.elastic4s.ElasticDsl._
+import org.scalatest.flatspec.AnyFlatSpec
+import com.sksamuel.elastic4s.cats.effect.instances._
+import com.sksamuel.elastic4s.http.JavaClient
+import com.sksamuel.elastic4s.requests.indexes.admin.DeleteIndexResponse
+import com.sksamuel.elastic4s.requests.indexes.{CreateIndexRequest, CreateIndexResponse, DeleteIndexRequest}
+
+class CatsEffectTest extends AnyFlatSpec {
```

Comment:
@sksamuel basically, without the changes in `CatsEffectInstances` the first test case would not compile unless we rewrite it with explicit type params like the second test case.

Reply:
Do users of standard scala futures have any breaking changes ?

Reply:
If users import `com.sksamuel.elastic4s.cats.effect.instances._` they intend `IO` to be the default effect. `Futures` should still work with explicit type params even with Cats Effect instances imported. I can add another test for that.

Reply:
Ok great. I just wanted to confirm that people who are not explicitly using effects would still be getting the default Futures implementation.

Reply:
@sksamuel Test cases updated and branch rebased. I will need approval to trigger CI.